### DISCUSSION TITLE
Error Handling?

### DISCUSSION THREAD

**Discussion by osseonews** [honojs/hono#2427] [Q&A]
Sorry for the newbie question, but how is error handling handled in Hono? I don't see any documentation on it, other than that you can have an app.onError method somewhere. Is this required to throw an error response? Where can this be in the hono file? My issue really is when developing custom middleware you can't return anything in middleware, so how do you return a 400 response from the middlware? If you throw an error, how does it propogate?

**Comment by cjfff** [comment]
Buddy, maybe you can refer to the koajs's solution, because it's the same structure framework alternative. https://github.com/koajs/koa/blob/master/docs/error-handling.md
